README.md

```diff
@@ -7,290 +7,505 @@
+
+To grant the scopes needed: 
+
+1. Go to [APIs](https://manage.auth0.com/#/apis) > Auth0 Management API > **Machine to Machine Applications** tab
+2. Find your Application and authorize it
+3. Click the arrow to expand the row and select the scopes required
+
+Now you can authenticate one of the two ways above and use that token to perform operations:
+
+```php
+use Auth0\SDK\API\Management;
+
+$access_token = getenv('AUTH0_MANAGEMENT_API_TOKEN');
+if ( empty( $access_token ) ) {
+	// See "Client Credentials Grant" above
+	$access_token = get_access_token();
+}
+$mgmt_api = new Management( $access_token, getenv('AUTH0_DOMAIN') );
 ```
 
-You will get your app client and secret from your Auth0 app you had created.
-The auth0 domain, is the one you pick when you created your auth0 account.
-You need to set this callback url in your app allowed callbacks.
-The app token is used in the 'create user' page and needs `create:users` scope. To create one, you need to use the token generator in the [API V2 documentation page](https://auth0.com/docs/apiv2)
+The `Management` class stores access to endpoints as properties of its instances. The best way to see what endpoints are covered is to read through the `\Auth0\SDK\API\Management::__construct()` method. 
 
-To run the example, you need composer (the PHP package manager) installed (you can find more info about composer [here](https://getcomposer.org/doc/00-intro.md)) and run the following commands on the same folder than the code.
+### Example - Search Users by Email
+
+This endpoint is documented [here](https://auth0.com/docs/api/management/v2#!/Users/get_users).
+
+```php
+$results = $mgmt_api->users->search([
+    'q' => 'josh'
+]);
 
+if (! empty($results)) {
+    echo '<h2>User Search</h2>';
+    foreach ($results as $datum) {
+        printf(
+            '<p><strong>%s</strong> &lt;%s&gt; - %s</p>',
+            !empty($datum['nickname']) ? $datum['nickname'] : 'No nickname',
+            !empty($datum['email']) ? $datum['email'] : 'No email',
+            $datum['user_id']
+        );
+    }
+}
 ```
-$ composer install
-$ php -S localhost:3000
+
+### Example - Get All Clients
+
+This endpoint is documented [here](https://auth0.com/docs/api/management/v2#!/Clients/get_clients).
+
+```php
+$results = $mgmt_api->clients->getAll();
+
+if (! empty($results)) {
+    echo '<h2>Get All Clients</h2>';
+    foreach ($results as $datum) {
+        printf(
+            '<p><strong>%s</strong> - %s</p>',
+            $datum['name'],
+            $datum['client_id']
+        );
+    }
+}
 ```
 
-## Migration guide
+## Contributing
 
-### from 1.x
+The SDKs we provide and maintain are for the benefit of our developer community so feedback, detailed bug reports, and focused PRs are all appreciated. Thank you in advance!
```

Comment:
Suggested change: We provide and maintain SDKs for the benefit of our developer community. Feedback, detailed bug reports, and focused PRs are appreciated. Thank you in advance!